File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/go_solutions/LCR 176-ping-heng-er-cha-shu-lcof.go
import "math"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isBalanced(root *TreeNode) bool {
	if root == nil {
		return true
	}
	height := func(node *TreeNode) int {
		if node == nil {
			return 0
		}
		return 1 + int(math.Max(float64(height(node.Left)), float64(height(node.Right))))
	}

	if math.Abs(float64(height(root.Left)-height(root.Right))) > 1 {
		return false
	}

	return isBalanced(root.Left) && isBalanced(root.Right)
}

File: leetcode/AIcoding/5.Qwen-3/java_solutions/LCR 176-ping-heng-er-cha-shu-lcof.java
public class Solution {
    public boolean isBalanced(TreeNode root) {
        return height(root) != -1;
    }

    private int height(TreeNode node) {
        if (node == null) {
            return 0;
        }
        int left = height(node.left);
        if (left == -1) {
            return -1;
        }
        int right = height(node.right);
        if (right == -1 || Math.abs(left - right) > 1) {
            return -1;
        }
        return 1 + Math.max(left, right);
    }

    static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
}